src/Craftwork/IdObfuscator/CharacterSet.php

```diff
@@ -0,0 +1,131 @@
+<?php
+declare(strict_types=1);
+
+namespace Craftwork\IdObfuscator;
+
+use Craftwork\IdObfuscator\Exception\DuplicateCharacterException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetException;
+use Craftwork\IdObfuscator\Exception\InvalidCharacterSetLengthException;
+
+final class CharacterSet
+{
+    const CHARSET_DEFAULT = 'abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ1234567890-_';
+
+    /**
+     * @var string
+     */
+    private $characters;
+
+    /**
+     * @param string $characters
+     */
+    private function __construct(string $characters)
+    {
+        $this->assertAllAsciiCharacters($characters);
+        $this->assertCharactersLength($characters);
+        $this->assertNoDuplicateCharacters($characters);
+
+        $this->characters = $characters;
+    }
+
+    /**
+     * @param string $characters
+     *
+     * @return CharacterSet
+     *
+     * @throws InvalidCharacterSetLengthException
+     * @throws DuplicateCharacterException
+     * @throws InvalidCharacterSetException
+     */
+    public static function ofCustomCharacters(string $characters): CharacterSet
+    {
+        return new self($characters);
+    }
+
+    /**
+     * @return CharacterSet
+     */
+    public static function ofDefaultCharacters(): CharacterSet
+    {
+        return new self(self::CHARSET_DEFAULT);
+    }
+
+    /**
+     * @return string
+     */
+    public function getCharacters(): string
+    {
+        return $this->characters;
+    }
+
+    /**
+     * @param string $salt
+     * @return CharacterSet
+     */
+    public function shuffleCharacterSetWithSalt(string $salt): CharacterSet
+    {
+        $saltLength = strlen($salt);
+
+        if ($saltLength === 0) {
```

Comment:
You didn't move this over to yoda (not that I'm a fan, unit tests and code reviews will pick up any bugs from doing assignment instead of comparison, and PHPStorm gives you a warning if you assign in an if statement anyway).